ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/writers/TomlEmitter.kt

```diff
@@ -198,20 +197,31 @@ public abstract class TomlEmitter(config: TomlOutputConfig) {
      * @param representation How the integer will be represented in TOML.
      * @return this instance
      */
-    public fun emitValue(integer: Long, representation: IntegerRepresentation = DECIMAL): TomlEmitter =
+    @Suppress("MAGIC_NUMBER")
+    public fun emitValue(integer: Long, representation: IntegerRepresentation = DECIMAL): TomlEmitter {
+        var value = integer
+
+        if (value < 0) {
+            return emit('-').emitValue(-value, representation)
+        }
```

Comment:
If we have Long.MIN_VALUE we get a StackOverflow error because 
`-Long.MIN_VALUE == Long.MIN_VALUE`

Reply:
Probably we could convert number to string first and then add grouping. Or any other idea :)

Reply:
Not giving up on a math-based solution just yet :P

Reply:
I just think, if we'll add `groupSize` in the future, is it easier to work with string?

Reply:
Maybe